driver/src/main/java/org/neo4j/driver/types/Entity.java

```diff
@@ -28,12 +28,22 @@
 public interface Entity extends MapAccessor
 {
     /**
-     * A unique id for this Entity. Ids are guaranteed to remain stable for the duration of the session they
-     * were found in, but may be re-used for other entities after that. As such, if you want a public identity to use
-     * for your entities, attaching an explicit 'id' property or similar persistent and unique identifier is a better
-     * choice.
+     * A unique id for this Entity. Ids are guaranteed to remain stable for the duration of the session they were found in, but may be re-used for other
+     * entities after that. As such, if you want a public identity to use for your entities, attaching an explicit 'id' property or similar persistent and
+     * unique identifier is a better choice.
      *
      * @return the id of this entity
+     * @deprecated superseded by {@link #elementId()}
      */
+    @Deprecated
     long id();
+
+    /**
+     * A unique id for this Entity. Ids are guaranteed to remain stable for the duration of the session they were found in, but may be re-used for other
```

Comment:
Is the same still true for element Id?

Reply:
also shall we add a note to `id()` to say a value of `-1` indicates unsupported or something util we work it out

Reply:
Updated.